app/models/backup_file.rb

```diff
@@ -16,6 +16,18 @@ def ==(other)
     attributes == other.attributes
   end
 
+  def self.download(url)
```

Comment:
Maybe `BackupRestore::BackupFileHandler.download(url)` instead? Putting it in the model doesn't seem like the right place.